hoodie-client/src/main/java/com/uber/hoodie/io/compact/HoodieRealtimeTableCompactor.java

```diff
@@ -215,11 +217,11 @@ public HoodieCompactionPlan generateCompactionPlan(JavaSparkContext jsc,
     // compactions only
     HoodieCompactionPlan compactionPlan = config.getCompactionStrategy().generateCompactionPlan(config, operations,
         CompactionUtils.getAllPendingCompactionPlans(metaClient).stream().map(Pair::getValue).collect(toList()));
-    Preconditions.checkArgument(compactionPlan.getOperations().stream()
-            .filter(op -> fileIdsWithPendingCompactions.contains(op.getFileId())).count() == 0,
+    Preconditions.checkArgument(compactionPlan.getOperations().stream().noneMatch(
+        op -> fgIdsInPendingCompactions.contains(new HoodieFileGroupId(op.getPartitionPath(), op.getFileId()))),
```

Comment:
should `op` just store a HoodieFileGroupId instead?

Reply:
Fixed.